avo-hq/avo
# Enable showing distribution on columns on index page

# Description

Fixes  [# 2162](https://github.com/avo-hq/avo/issues/2162)

The issue describes the feature in a general way, so I've applied it in my way, although according to our discussion on the Issue.

From the things worth mentioning - I've decided not to implement handling of multiple "widgets" on one column, eg. `sortable` and `summarizable`. In my opinion we can do it as a second iteration and separate task. Let me know if you thing otherwise @adrianthedev. 

When it comes to the colors - I've noticed that although we are defining the `default_chart_colors` and `chart_colors` (on `Avo::Configuration::Branding`), we're not using it anywhere (or at least I haven't found the usage of it).
For this reason, I've decided to change the `default_chart_colors` to match the choice of the `chartkick` lib that we're using. 
In their case you can find the defaults [here](https://github.com/ankane/chartkick/blob/master/vendor/assets/javascripts/chartkick.js#L401) and I think they have a valid reasoning behind the list.

It will be possible to customize the colors for charts, although in that case user will have to provide the arbitrary classes for the Tailwind (eg. in the `safelist.txt`).


## Screenshots & recording

Updated after code review - 24.02.2024 (what a date!)

https://github.com/avo-hq/avo/assets/12682792/839c7ee8-787a-4683-98c3-ca0097db8536


## Manual review steps

1. Run the dummy app and check the Project resource index page
1. Check the behavior when clicking on the bar icon near the "Status" table header

Manual reviewer: please leave a comment with output from the test if that's the case.
